roles/reboot/tasks/main.yml

```diff
@@ -22,24 +22,10 @@
 # that they don't use sudo, which may require a password, and is not necessary
 # anyway.
 
-# During playbook testing, I found that sometimes the 'ansible_host' variable
-# is sometimes undefined, leading to errors waiting for the host to return.
-# We can opt to use the 'inventory_hostname' in this case and avoid those
-# (admittedly) rare error cases.
-- name: wait for hosts to come back up (ansible_host)
-  local_action: wait_for host={{ ansible_host }}
-                port=22
-                state=started
-                delay=15
-                timeout=120
-  become: false
-  when: ansible_host is defined
-
 - name: wait for hosts to come back up (inventory_hostname)
   local_action: wait_for host={{ inventory_hostname }}
```

Comment:
The problems we have observed are when `ansible_host` is defined, so I don't want to use it at all, if possible.

Like you said `inventory_hostname` is always defined (AFAICT) and is reliable, so let's just use it outright.